rest/rest.go

```diff
@@ -45,16 +48,91 @@
 	"google.golang.org/grpc/credentials/insecure"
 )
 
-var log *logrus.Entry
+var (
+	log *logrus.Entry
+
+	// srv holds the global http.Server of our REST API.
+	srv *http.Server
+
+	// sock holds the listener socket of our REST API.
+	sock net.Listener
+
+	ready = make(chan bool)
+
+	// cors holds the global CORS configuration.
+	cors *corsConfig
+)
+
+// corsConfig holds all necessary configuration options for Cross-Origin Resource Sharing of our REST API.
+type corsConfig struct {
+	// allowedOrigins contains a list of the allowed origins
+	allowedOrigins []string
+
+	// allowedHeaders contains a list of the allowed headers
+	allowedHeaders []string
+
+	// allowedMethods contains a list of the allowed methods
+	allowedMethods []string
+}
+
+// CORSConfigOption represents functional-style options to modify the CORS configuration in RunServer.
+type CORSConfigOption func(*corsConfig)
+
+var (
+	// DefaultAllowedOrigins contains an nil slice, as per default, no origins are allowed.
+	DefaultAllowedOrigins []string = nil
+
+	// DefaultAllowedHeaders contains sensible defaults for the Access-Control-Allow-Headers header.
+	// Please adjust accordingly in production using WithAllowedHeaders.
+	DefaultAllowedHeaders = []string{"Content-Type", "Accept", "Authorization"}
+
+	// DefaultAllowedMethods contains sensible defaults for the Access-Control-Allow-Methods header.
+	// Please adjust accordingly in production using WithAllowedMethods.
```

Comment:
Is it ok to allow POST, PUT and DELETE methods as default as they are changing stuff? Or is it enough since we block allowed origins already per default?

Reply:
I think it should be ok, as we are blocking origins per default.